fms/modules/feedforward.py

```diff
@@ -289,30 +335,55 @@ def load_weights(
     ):
         # 1. Grab the weights from tensor_values
         used_keys: Set[str] = set()
-        wg_weight = self._get_sd_weight(
-            tensor_values, used_keys, ["wg1_fused", "weight"]
-        )
+        if self.fused:
+            wg_weight = self._get_sd_weight(
+                tensor_values, used_keys, ["wg1_fused", "weight"]
+            )
+        else:
```

Comment:
are we ever going to apply TP after unfusing? Don't we generally do get_model first and then unfuse?

Reply:
It's possible that a model will create GLU with fuse=False, in which case we will need to tp-shard the unfused module. Currently we don't have any models that do this, but it is a possibility

Reply:
ok, understood